src/coreComponents/physicsSolvers/fluidFlow/DissipationCompositionalMultiphaseFVMKernels.hpp

```diff
@@ -0,0 +1,472 @@
+                              fields::multifluid::phaseMassDensity,
+                              fields::multifluid::phaseCompFraction_n,
+                              fields::multifluid::phaseViscosity >;
+
+  using RelPermAccessors =
+    StencilMaterialAccessors< RelativePermeabilityBase, fields::relperm::phaseRelPerm_n >;
+
+  using PorosityAccessors =
+    StencilMaterialAccessors< PorosityBase, fields::porosity::porosity_n >;
+
+  using AbstractBase::m_dt;
+  using AbstractBase::m_numPhases;
+  using AbstractBase::m_hasCapPressure;
+  using AbstractBase::m_rankOffset;
+  using AbstractBase::m_ghostRank;
+  using AbstractBase::m_dofNumber;
+  using AbstractBase::m_gravCoef;
+  using AbstractBase::m_phaseMob;
+  using AbstractBase::m_dPhaseMassDens;
+  using AbstractBase::m_phaseCompFrac;
+  using AbstractBase::m_dPhaseCompFrac;
+  using AbstractBase::m_dCompFrac_dCompDens;
+  using AbstractBase::m_dPhaseCapPressure_dPhaseVolFrac;
+  using AbstractBase::m_pres;
+
+  using Base = isothermalCompositionalMultiphaseFVMKernels::FaceBasedAssemblyKernel< NUM_COMP, NUM_DOF, STENCILWRAPPER >;
+  using Base::numComp;
+  using Base::numDof;
+  using Base::numEqn;
+  using Base::numFluxSupportPoints;
+  using Base::maxNumElems;
+  using Base::maxNumConns;
+  using Base::maxStencilSize;
+  using Base::m_stencilWrapper;
+  using Base::m_seri;
+  using Base::m_sesri;
+  using Base::m_sei;
+
+  using Deriv = multifluid::DerivativeOffset;
+
+  /**
+   * @brief Constructor for the kernel interface
+   * @param[in] numPhases the number of fluid phases
+   * @param[in] rankOffset the offset of my MPI rank
+   * @param[in] hasCapPressure flag specifying whether capillary pressure is used or not
+   * @param[in] stencilWrapper reference to the stencil wrapper
+   * @param[in] dofNumberAccessor accessor for the dofs numbers
+   * @param[in] compFlowAccessor accessor for wrappers registered by the solver
+   * @param[in] dissCompFlowAccessor accessor for wrappers registered by the solver needed for dissipation
+   * @param[in] multiFluidAccessor accessor for wrappers registered by the multifluid model
+   * @param[in] dissMultiFluidAccessor accessor for wrappers registered by the multifluid model needed for dissipation
+   * @param[in] capPressureAccessors accessor for wrappers registered by the cap pressure model
+   * @param[in] permeabilityAccessors accessor for wrappers registered by the permeability model
+   * @param[in] porosityAccessors accessor for wrappers registered by the porosity model
+   * @param[in] relPermAccessors accessor for wrappers registered by the relative permeability model needed for dissipation
+   * @param[in] dt time step size
+   * @param[inout] localMatrix the local CRS matrix
+   * @param[inout] localRhs the local right-hand side vector
+   */
+  FaceBasedAssemblyKernel( integer const numPhases,
+                           globalIndex const rankOffset,
+                           integer const hasCapPressure,
```

Comment:
i am pretty sure that some of params can be dropped, e.g. related to cap pressure